include/exadg/operators/operator_base.h

```diff
@@ -715,13 +716,6 @@ class OperatorBase : public dealii::Subscriptor
    */
   mutable VectorType weights;
 
-  /*
-   * We want to initialize the block diagonal preconditioner (block diagonal matrices or elementwise
-   * iterative solvers in case of matrix-free implementation) only once, so we store the status of
-   * initialization in a variable.
-   */
-  mutable bool block_diagonal_preconditioner_is_initialized;
```

Comment:
Before, we had this logic where we potentially initialize when calling `update`. So, the preconditioner was initialized when being updated. Now, we initialize in the constructor and later only update, hence do not need to keep track of the state using `preconditioner_is_initialized`, checking this variable and potentially initialize.

Two questions here:
i) Is the `underlying_operator` in all cases ready at the time we setup the preconditioner following the current strategy (remember, we were discussing the scaling factors and so on)? Before this PR, there *could* -- I am not saying there were, but there might be -- some changes to the `underlying_operator` inbetween preconditioner setup and calling the former `update` including the initialization step.
ii) In the parallel PR #599, we actually introduce the strategy of initializing when calling update if necessary (what we had here before this PR). I think we should keep this aspect of the previous implementation, but **do split** into `update` and `initialize` steps as done here. In your opinion, would this only needlessly complicate matters here, since it is always clear that the preconditioner has been initialized after calling the constructor, right?

These questions relate to the differences between the preconditioner and the solver and why we go different routes within ExaDG for each of these. It would be more flexible, if we would not need the `underlying_operator` to be initialized when constructing the preconditioner. The question is, if this is overengineered was introduced previously for a good reason?

Reply:
I did not fully understand this. Where do you think did something change in the logic?

So far, we need the underlying operator to be initialized before creating/initializing the preconditioner. I think the present PR does not change this, and also should not change this because this is to be addressed separately.

Reply:
The logic isnt changed in this PR, but we had that variable `preconditioner_is_initialized`, which would allow similar logic as is the goal in #602, isn't it? Wouldn't that mean we remove the variable here and introduce it again later?